plugins/Goals/DataTable/Filter/CalculateConversionPageRate.php

```diff
@@ -0,0 +1,119 @@
+        parent::__construct($table);
+    }
+
+    /**
+     * @param DataTable $table
+     */
+    public function filter($table)
+    {
+
+        // Find all goal ids used in the table and store in an array
+        $goals = [];
+        foreach ($table->getRowsWithoutSummaryRow() as $row) {
+            if (isset($row[Metrics::INDEX_GOALS])) {
+                foreach ($row[Metrics::INDEX_GOALS] as $goalIdString => $metrics) {
+                    $goals[$goalIdString] = $goalIdString;
+                }
+            }
+        }
+
+        // Get the total top-level conversions for the goals in the table
+        $goalTotals = $this->getGoalTotalConversions($table, $goals);
+        if (count($goalTotals) === 0) {
+            return;
+        }
+
+        // Walk the rows and populate the nb_conversions_page_rate with nb_conversions_page_uniq / $goalTotals[goal id]
+        foreach ($table->getRowsWithoutSummaryRow() as &$row) {
+            if (isset($row[Metrics::INDEX_GOALS])) {
+                foreach ($row[Metrics::INDEX_GOALS] as $goalIdString => $metrics) {
+                    if (isset($row[Metrics::INDEX_GOALS][$goalIdString][Metrics::INDEX_GOAL_NB_CONVERSIONS_PAGE_UNIQ])) {
+
+                        $rate = Piwik::getQuotientSafe(
+                                $row[Metrics::INDEX_GOALS][$goalIdString][Metrics::INDEX_GOAL_NB_CONVERSIONS_PAGE_UNIQ],
+                                $goalTotals[$goalIdString],
+                                3
+                            );
+                        // Prevent page rates over 100% which can happen when there are subpages
+                        if ($rate > 1) {
+                            $rate = 1;
+                        }
+
+                        $row[Metrics::INDEX_GOALS][$goalIdString][Metrics::INDEX_GOAL_NB_CONVERSIONS_PAGE_RATE] = $rate;
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Get the conversions total for each goal in the top level datatable
+     *
+     * @param DataTable $table
+     * @param array $goalIds
+     * @return array
+     */
+    private function getGoalTotalConversions(DataTable $table, array $goalIds)
+    {
+
+        $goalTotals = [];
+
+        /** @var Site $site */
+        $site = $table->getMetadata('site');
+        if (empty($site)) {
+            return $goalTotals;
+        }
+        $idSite = $site->getId();
+
+        $period = $table->getMetadata('period');
+        $periodName = $period->getLabel();
+        $date = $period->getDateStart()->toString();
+
+        foreach ($goalIds as $idGoal => $g) {
+
+            $params = [
+                'period' => $periodName,
+                'date' => ($periodName === 'range' ? $date.','.$period->getDateEnd()->toString() : $date),
+                'idSite' => $idSite,
+                'idGoal' => $idGoal
+            ];
+
+            /** @var DataTable $data */
+            $data = Request::processRequest( 'Goals.get', $params);
```

Comment:
Depending on the intention you should set the third parameter to an empty array. Otherwise all request parameters from the original requests will be used as default. This would include the segment, but also some filter parameters, that might hide required columns.